jest.preset.js

```diff
@@ -0,0 +1,28 @@
+// @ts-check
+
+const path = require('path');
+const { pathsToModuleNameMapper } = require('ts-jest/utils');
+
+const tsConfig = require('./tsconfig.base.json');
+
+const tsPathAliases = pathsToModuleNameMapper(tsConfig.compilerOptions.paths, {
+  prefix: `<rootDir>/${path.relative(process.cwd(), __dirname)}/`,
+});
+
+/**
+ * @type {jest.InitialOptions}
+ */
+const baseConfig = {
+  transform: {
+    '^.+\\.ts$': 'ts-jest',
+  },
+  testMatch: ['**/+(*.)+(spec|test).+(ts|js)?(x)'],
+  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
+  testPathIgnorePatterns: ['/node_modules/', '/lib/', '/lib-commonjs/'],
+  moduleNameMapper: { ...tsPathAliases },
+  cacheDirectory: '<rootDir>/.cache/jest',
+  clearMocks: true,
+  watchPlugins: ['jest-watch-typeahead/filename', 'jest-watch-typeahead/testname'],
+};
+
+module.exports = { ...baseConfig };
```

Comment:
Not very important but why the spread?

Reply:
extensibility without less git diff noise.

my thinking was to keep it like this -> extract common preset to separate package with tests etc (not within scripts/), then just use api from that package and leave room for any tweaks additions in here without noisy git diffs